packages/components/src/toggle-group-control/toggle-group-control-option-base/component.tsx

```diff
@@ -142,19 +140,17 @@ function ToggleGroupControlOptionBase(
 				) : (
 					<Ariakit.Radio
 						disabled={ disabled }
-						render={
-							<button
-								type="button"
-								{ ...commonProps }
-								onFocus={ ( event ) => {
-									onFocusProp?.( event );
-									if ( event.defaultPrevented ) {
-										return;
-									}
-									toggleGroupControlContext.setValue( value );
-								} }
-							/>
-						}
+						onFocusVisible={ () => {
+							// Conditions ensure that the first visible focus to a radio group
+							// without a selected option will not automatically select the option.
+							if (
+								toggleGroupControlContext.value !== null ||
+								toggleGroupControlContext.activeItemIsNotFirstItem?.()
```

Comment:
This `activeItemIsNotFirstItem()` check will only trigger once in the toggle group's lifecycle, as it will have a non-null `value` in subsequent renders.

Reply:
I'm not sure I understand the logic here — if the current `value` is `null`, why would we call `setValue` if the active item is not the first item?

Reply:
The logic in trunk was that, whenever an option is focused (i.e. becomes the active item), set that value in the Ariakit store explicitly (because we're managing this component in controlled mode). We want to prevent this behavior when there is no value selected yet and the active item is the first item. Once the active item is the second item, that means the user not only "focused on the ToggleGroupControl" but also actively selected an option, so we are safe to start setting values. This is how native radios behave.

Hope that made better sense, but also I'd like your input on how to improve the code comment so it is easier to understand.

Reply:
Is the `toggleGroupControlContext.activeItemIsNotFirstItem` check necessary, though? Does it matter which item is the active item?

Reply:
Yes. See for example the default example for [Ariakit Radio](https://ariakit.org/components/radio). When the focus first enters the radio group, the active item is the first item ("apple"). It's only when the active item becomes something else ("orange") that we want to start setting values.

https://github.com/user-attachments/assets/1190d90c-96b1-4f25-b58a-b4e94816505e

Once we start setting values (i.e. the `value` is non-null), then it no longer matters which is the active item.

Reply:
Got it. I wish we could rely on the built-in behavior of the ariakit `Radio` component, without the need for us to re-implement the logic (I guess we can't since we're not using native `input type="radio"` elements?)

Also, if we even want to tweak the initial position of the active element before a selection is made, this solution won't cut it - but it's unlikely, so that doesn't seem like a big issue

Reply:
Sounds good — we may want to consider a v2 when we have the new design spec